recipes/_hadoop_hdfs_ha_checkconfig.rb

```diff
@@ -27,6 +27,9 @@
 # We have dfs.nameservices, and now need to check them
 dfs_nameservices.each do |ns|
   # Start namenode checks
+  if node['hadoop']['hdfs_site']['dfs.ha.namenodes'] 
+    Chef::Application.fatal!("NameNode HA requires node['hadoop']['hdfs_site']['dfs.ha.namenodes.#{ns}']") unless node['hadoop']['hdfs_site'].key?("dfs\.ha\.namenodes\.#{ns}")
+  end
```

Comment:
Put another way it says, "Hey you set  dfs.nameservers and you set
dfs.ha.namenodes but don't have dfs.ha.namenodes.NameServer"

It silently failed/continued on me.

On Thu, Jul 2, 2015 at 4:13 PM, Anthony Acquanita <anthony@acquanita.me>
wrote:

> Following this trying to setup a HA name node with journal nodes
>
> https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-hdfs/HDFSHighAvailabilityWithQJM.html
>
> Using the cookbook it converged but I noticed the hdfs-site config didn't
> have the settings below even though I intended to do HA by setting
> hdfs.ha.namenodes and nothing else.
>
> *dfs.nameservices* - the logical name for this new nameservice
> *dfs.ha.namenodes.[nameservice ID]* - unique identifiers for each
> NameNode in the nameservice
> ​*dfs.namenode.rpc-address.[nameservice ID].[name node ID]* - the
> fully-qualified RPC address for each NameNode to listen on
> *dfs.namenode.http-address.[nameservice ID].[name node ID]* - the
> fully-qualified HTTP address for each NameNode to listen on​
> ​*dfs.namenode.shared.edits.dir* - the URI which identifies the group of
> JNs where the NameNodes will write/read edits
> *dfs.client.failover.proxy.provider.[nameservice ID]* - the Java class
> that HDFS clients use to contact the Active NameNode
> *dfs.ha.fencing.methods* - a list of scripts or Java classes which will
> be used to fence the Active NameNode during a failover
> *fs.defaultFS* - the default path prefix used by the Hadoop FS client
> when none is given
> *dfs.journalnode.edits.dir* - the path where the JournalNode daemon will
> store its local state​
>
>
> On Thu, Jul 2, 2015 at 4:05 PM, Chris Gianelloni <notifications@github.com
> > wrote:
>
>> In recipes/_hadoop_hdfs_ha_checkconfig.rb
>> <https://github.com/caskdata/hadoop_cookbook/pull/209#discussion_r33817066>
>> :
>>
>> > @@ -27,6 +27,9 @@
>> >  # We have dfs.nameservices, and now need to check them
>> >  dfs_nameservices.each do |ns|
>> >    # Start namenode checks
>> > +  if node['hadoop']['hdfs_site']['dfs.ha.namenodes']
>> > +    Chef::Application.fatal!("NameNode HA requires node['hadoop']['hdfs_site']['dfs.ha.namenodes.#{ns}']") unless node['hadoop']['hdfs_site'].key?("dfs\.ha\.namenodes\.#{ns}")
>> > +  end
>>
>> These aren't checking the same thing. What are you actually trying to do
>> here?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/caskdata/hadoop_cookbook/pull/209/files#r33817066>.
>>
>
>